homeassistant/components/statistics/sensor.py

```diff
@@ -170,15 +177,27 @@ def valid_state_characteristic_configuration(config: dict[str, Any]) -> dict[str
     return config
 
 
+def valid_boundary_configuration(config: dict[str, Any]) -> dict[str, Any]:
+    """Validate that either sampling_size or max_age are provided."""
+
+    if config.get(CONF_SAMPLES_MAX_BUFFER_SIZE) is None:
+        config[CONF_SAMPLES_MAX_BUFFER_SIZE] = DEFAULT_BUFFER_SIZE
```

Comment:
Should we take the plunge and only set the default buffer size if `CONF_MAX_AGE` is not set, although that would be a breaking change?

Reply:
Would love to but beats the purpose of the deprecation warning. Users with implicit buffer size will be asked to make it explicit today. When I add the 'breaking change PR' later, no users are left to experience the break... win-win

Quite the contrary, I think there are many user who have defined a ridiculous buffer size like 99999 in their config, just to work around the current limitations. I would love to inform them that they can remove that silly line at this point... hmmm